staticaddr/loopin/selected_amount_migration.go

```diff
@@ -0,0 +1,99 @@
+package loopin
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/lightninglabs/loop/loopdb"
+	"github.com/lightninglabs/loop/staticaddr/deposit"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+const (
+	// selectedAmountMigrationID is the identifier for the selected swap
+	// amount migration.
+	selectedAmountMigrationID = "selected_amount"
+)
+
+// MigrateSelectedSwapAmount will update the selected swap amount of past swaps
+// with the sum of the values of the deposits they swapped.
+func MigrateSelectedSwapAmount(ctx context.Context, db loopdb.SwapStore,
+	depositStore *deposit.SqlStore, swapStore *SqlStore) error {
+
+	migrationDone, err := db.HasMigration(ctx, selectedAmountMigrationID)
```

Comment:
Imo the whole migration should be done in a single transaction.

Reply:
The actual migration is done in a single batch in `BatchUpdateSelectedSwapAmounts`. Is that what you mean?